viewing_party/party.py

```diff
@@ -1,23 +1,165 @@
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    
+    sum = 0
+
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    
+    return sum / len(user_data["watched"])
+
+
+def get_most_watched_genre(user_data):
+
+    if len(user_data["watched"]) == 0:
+        return None
+
+    genres_count = {}
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genres_count:
+            genres_count[genre] += 1
+        else:
+            genres_count[genre] = 1
+    
+    max_count = 0
+    most_frequently_watched_genre = ""
+
+    for genre in genres_count.keys():
+        if genres_count[genre] > max_count:
+            max_count = genres_count[genre]
+            most_frequently_watched_genre = genre
+    
+    return most_frequently_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    unique_movies = []
+
+    for movie in user_data["watched"]:
+        is_unique = True
+        for friend in user_data["friends"]:
+            for friend_movie in friend["watched"]:
+                if movie["title"] == friend_movie["title"]:
+                    is_unique = False
+        
+        if is_unique:
+            unique_movies.append(movie)
+
+    return unique_movies
+
+
+def get_friends_unique_watched(user_data):
+    unique_movies = []
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            is_unique = True
+            for movie in user_data["watched"]:
+                if movie["title"] == friend_movie["title"]:
+                    is_unique = False
+        
+            if is_unique:
+                is_unique_and_new = True
+                for unique_movie in unique_movies:
+                    if unique_movie["title"] == friend_movie["title"]:
+                        is_unique_and_new = False
+
+                if is_unique_and_new:
+                    unique_movies.append(friend_movie)
+
+    return unique_movies
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended_movies = []
+
+    friends_unique_watched = get_friends_unique_watched(user_data)
+
+    for movie in friends_unique_watched:
+        for subscription in user_data["subscriptions"]:
+            if subscription == movie["host"]:
+                recommended_movies.append(movie)
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    most_watched_genre_movies = []

```

Comment:
This is just a small thing, but a slightly more descriptive variable here might be better. Maybe instead of using "movies", you can use "recs" to signify that what you are outputting are recommendations rather than just random movies!